pyproject.toml

```diff
@@ -114,11 +114,17 @@ dev = [
     # To access the metadata for python packages
     "importlib-metadata; python_version < '3.10'",
 ]
-# For the Jax solver. Note: these must be kept in sync with the versions defined in pybamm/util.py.
+# For the Jax solver.
+# Note: These must be kept in sync with the versions defined in pybamm/util.py, and
+#       must remain compatible with IREE (see noxfile.py for IREE compatibility).
 jax = [
     "jax==0.4.27",
     "jaxlib==0.4.27",
 ]
+# For MLIR expression evaluation (IDAKLU Solver)
+iree = [
+    "iree-compiler==20240507.886",  # see IREE compatibility notes in noxfile.py
+]
```

Comment:
Please add a note about this in the table of dependencies in the docs: https://docs.pybamm.org/en/latest/source/user_guide/installation/index.html and additionally in the Optional solvers section as well